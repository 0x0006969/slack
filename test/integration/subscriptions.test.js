const request = require('supertest');
const nock = require('nock');

const helper = require('.');
const fixtures = require('../fixtures');

describe('Integration: subscriptions', () => {
  beforeEach(async () => {
    // Create an installation
    await helper.robot.models.Installation.create({
      githubId: 1,
      ownerId: fixtures.org.id,
    });

    nock.cleanAll();
  });

  afterEach(() => {
    // Expect there are no more pending nock requests
    expect(nock.pendingMocks()).toEqual([]);
  });

  test('successfully subscribing to a repository', async () => {
    const { probot } = helper;

    nock('https://api.github.com').get('/orgs/atom').reply(200, fixtures.org);
    nock('https://api.github.com').get('/repos/atom/atom').reply(200, fixtures.repo);

    const command = fixtures.slack.command({
      text: 'subscribe https://github.com/atom/atom',
    });

    const req = request(probot.server).post('/slack/command').send(command);

    await req.expect(200).expect((res) => {
      expect(res.body).toMatchSnapshot();
    });
  });

  test('successfully subscribing with repository shorthand', async () => {
    const { probot } = helper;

    nock('https://api.github.com').get('/orgs/atom').reply(200, fixtures.org);
    nock('https://api.github.com').get('/repos/atom/atom').reply(200, fixtures.repo);

    const command = fixtures.slack.command({ text: 'subscribe atom/atom' });

    const req = request(probot.server).post('/slack/command').send(command);

    await req.expect(200).expect((res) => {
      expect(res.body).toMatchSnapshot();
    });
  });

  test('subsscribing with a bad url', async () => {
    const { probot } = helper;

    const command = fixtures.slack.command({ text: 'subscribe wat?' });

    const req = request(probot.server).post('/slack/command').send(command);

    await req.expect(200).expect((res) => {
      expect(res.body).toMatchSnapshot();
    });
  });
});
