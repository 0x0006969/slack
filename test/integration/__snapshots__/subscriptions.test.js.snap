// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration: subscriptions subsscribing with a bad url 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "danger",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "\`wat?\` does not appear to be a GitHub link.",
    },
  ],
}
`;

exports[`Integration: subscriptions successfully subscribing to a repository 1`] = `
Object {
  "attachments": Array [
    Object {
      "text": "Subscribed <#C2147483705> to <https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
    },
  ],
  "response_type": "in_channel",
}
`;

exports[`Integration: subscriptions successfully subscribing with repository shorthand 1`] = `
Object {
  "attachments": Array [
    Object {
      "text": "Subscribed <#C2147483705> to <https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
    },
  ],
  "response_type": "in_channel",
}
`;
